streams/shared/src/main/scala/zio/stream/experimental/ZTransducer.scala

```diff
@@ -124,6 +125,250 @@ object ZTransducer {
   def fromPush[R, E, I, O](push: Option[Chunk[I]] => ZIO[R, E, Chunk[O]]): ZTransducer[R, E, I, O] =
     ZTransducer(Managed.succeed(push))
 
+  /**
+   * Creates a transducer by folding over a structure of type `S`.
+   */
+  def fold[I, O](z: O)(contFn: O => Boolean)(f: (O, I) => (O, Chunk[I])): ZTransducer[Any, Nothing, I, O] =
+    ZTransducer {
+      final case class FoldState(started: Boolean, result: O, buffer: Chunk[I])
```

Comment:
Why do we need to buffer `I` values? Conceptually the operator can fold incoming chunks using `f` and whenever `contFn == false`, the resulting `O` is added to the output chunk. Multiple `O` values can be emitted in a single chunk. This way you'd only need to buffer an `O` value for which `contFn` was always true.

Reply:
Ah, right, I'll address that.